pkg/kubelet/dockershim/helpers_windows.go

```diff
@@ -83,7 +83,31 @@ func (ds *dockerService) updateCreateConfig(
 	return nil
 }
 
-func (ds *dockerService) determinePodIPBySandboxID(sandboxID string) string {
+func (ds *dockerService) determinePodIPBySandboxID(sandboxID string, sandbox *dockertypes.ContainerJSON) string {
+	// Versions and feature support
+	// ============================
+	// Windows version >= Windows Server, Version 1709, Supports both sandbox and non-sandbox case
+	// Windows version == Windows Server 2016 Support only non-sandbox case
+	// Windows version < Windows Server 2016 is Not Supported
+
+	// Sandbox support in Windows mandates CNI Plugin.
+	// Presence of CONTAINER_NETWORK flag is considered as non-Sandbox cases here
+	// Hyper-V isolated containers are also considered as non-Sandbox cases
+
+	// Todo: Add a kernel version check for more validation
```

Comment:
Could we change this loop to something more easy to read? e.g.
```go
if networkMode := os.Getenv("CONTAINER_NETWORK"); networkMode == "" && sandbox.HostConfig.Isolation != kubeletapis.HypervIsolationValue {
  return ds.getIP(sandboxID, sandbox)
}
```

Reply:
I can do that, but it will be harder to understand the comment afterwards which is referring only to the second part of the if.

Reply:
I think it's same, there's still hyperv check and the comments are also there

Reply:
Done.